mocket/inject.py

```diff
@@ -27,99 +37,62 @@ def enable(
+        (socket, "inet_pton"): mock_inet_pton,
+        (socket, "SocketType"): MocketSocket,
+        (socket, "socketpair"): mock_socketpair,
+        # stdlib: ssl
+        (ssl, "SSLContext"): MocketSSLContext,
+        (ssl, "wrap_socket"): mock_wrap_socket,  # python < 3.12.0
+        # urllib3
+        (urllib3.connection, "match_hostname"): mock_urllib3_match_hostname,
+        (urllib3.connection, "ssl_wrap_socket"): mock_urllib3_ssl_wrap_socket,
+        (urllib3.util, "ssl_wrap_socket"): mock_urllib3_ssl_wrap_socket,
+        (urllib3.util.ssl_, "ssl_wrap_socket"): mock_urllib3_ssl_wrap_socket,
+        (urllib3.util.ssl_, "wrap_socket"): mock_urllib3_ssl_wrap_socket,  # urllib3 < 2
+    }
+
+    for (module, name), new_value in patches.items():
+        _patch(module, name, new_value)
+
+    with contextlib.suppress(ImportError):
+        from urllib3.contrib.pyopenssl import extract_from_urllib3
+
+        extract_from_urllib3()
+
+    from mocket.mocket import Mocket
 
     Mocket._namespace = namespace
     Mocket._truesocket_recording_dir = truesocket_recording_dir
-
     if truesocket_recording_dir and not os.path.isdir(truesocket_recording_dir):
         # JSON dumps will be saved here
         raise AssertionError
 
-    socket.socket = socket.__dict__["socket"] = MocketSocket
-    socket._socketobject = socket.__dict__["_socketobject"] = MocketSocket
-    socket.SocketType = socket.__dict__["SocketType"] = MocketSocket
-    socket.create_connection = socket.__dict__["create_connection"] = (
-        mock_create_connection
-    )
-    socket.gethostname = socket.__dict__["gethostname"] = mock_gethostname
-    socket.gethostbyname = socket.__dict__["gethostbyname"] = mock_gethostbyname
-    socket.getaddrinfo = socket.__dict__["getaddrinfo"] = mock_getaddrinfo
-    socket.socketpair = socket.__dict__["socketpair"] = mock_socketpair
-    ssl.wrap_socket = ssl.__dict__["wrap_socket"] = MocketSSLContext.wrap_socket
-    ssl.SSLContext = ssl.__dict__["SSLContext"] = MocketSSLContext
-    socket.inet_pton = socket.__dict__["inet_pton"] = mock_inet_pton
-    urllib3.util.ssl_.wrap_socket = urllib3.util.ssl_.__dict__["wrap_socket"] = (
-        MocketSSLContext.wrap_socket
-    )
-    urllib3.util.ssl_.ssl_wrap_socket = urllib3.util.ssl_.__dict__[
-        "ssl_wrap_socket"
-    ] = MocketSSLContext.wrap_socket
-    urllib3.util.ssl_wrap_socket = urllib3.util.__dict__["ssl_wrap_socket"] = (
-        MocketSSLContext.wrap_socket
-    )
-    urllib3.connection.ssl_wrap_socket = urllib3.connection.__dict__[
-        "ssl_wrap_socket"
-    ] = MocketSSLContext.wrap_socket
-    urllib3.connection.match_hostname = urllib3.connection.__dict__[
-        "match_hostname"
-    ] = mock_urllib3_match_hostname
-    if pyopenssl_override:  # pragma: no cover
-        # Take out the pyopenssl version - use the default implementation
-        extract_from_urllib3()
-
 
 def disable() -> None:
+    for module, name in list(_patches_restore.keys()):
```

Comment:
I don't see the need for casting it to a `list`:

```suggestion
    for module, name in _patches_restore.keys():
```

Reply:
as `_restore` removes elements from `_patch_restore` the keys have to be converted to list here as to prevent `RuntimeError: dictionary changed size during iteration`

Reply:
Makes sense, thank you!